### ISSUE TITLE
Method not found: CommitAsync

### ISSUE THREAD

**Issue by eltiare** [jeff-pang/Nancy.AspNetCore.Session#2]
It seems that the interface for IDistributedCache has changed  between 1.6 and 2.0 of .NET Standard. I'm running .NET core 2.0 and I get the following error when using the distributed memory cache: 

```
Unhandled Exception: System.MissingMethodException: Method not found: 'System.Threading.Tasks.Task Microsoft.AspNetCore.Http.ISession.CommitAsync()'.
   at Nancy.AspNetCore.Session.NancyAspNetCoreSession.<SaveAsync>d__16.MoveNext()
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[TStateMachine](TStateMachine& stateMachine)
   at Nancy.AspNetCore.Session.NancyAspNetCoreSession.SaveAsync()
   at Nancy.AspNetCore.Session.NancyAspNetCoreSession.<SaveSession>d__23.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Nancy.AspNetCore.Session.NancyAspNetCoreSession.<>c.<<Enable>b__21_1>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
```

Since I'm still fairly new at all of this I am not sure how to proceed to fix this.

**Comment by eltiare** [comment]
It's reporting on line 201, which (just in case I modified the project locally) is the second line in the function here: 

```
public static void Enable(IPipelines pipelines,IHttpContextAccessor httpCtxAcs)
        {
            pipelines.BeforeRequest.AddItemToStartOfPipeline(ctx => LoadSession(ctx, httpCtxAcs.HttpContext));
            pipelines.AfterRequest.AddItemToEndOfPipeline(async ctx => await SaveSession(ctx));
        }
```

**Comment by eltiare** [comment]
I wanted to make sure it wasn't something that I had done, so I made an example here: 

https://github.com/eltiare/nancy-netcore-session-bug

**Comment by eltiare** [comment]
I just downgraded the .net core packages and environment to 1.1. It works there. Oy. :(

**Comment by eltiare** [comment]
This is now fixed with the 2.0 release.
